Mathlib/LinearAlgebra/LinearProgramming.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Antoine Chambert-Loir
+-/
+import Mathlib.Analysis.Convex.Cone.Pointed
+import Mathlib.Analysis.Convex.Topology
+import Mathlib.Analysis.NormedSpace.HahnBanach.Separation
+import Mathlib.Algebra.Order.Group.Defs
+import Mathlib.RingTheory.Finiteness
+
+/-!
+
+# Linear programming
+
+TODO
+
+-/
+
+/-- Typically `M` is `ℝ^m` and `N` is `ℝ^n` -/
+structure LinearProgram (R : Type*) (M : Type*) (N : Type*)
+    [OrderedSemiring R] [AddCommMonoid M] [Module R M] [AddCommMonoid N] [Module R N] where
```

Comment:
I think it _probably_ makes sense to generalize M to an affine space, though I guess you'd be trading no negation for no origin, as we don't have a typeclass that means neither...

Reply:
Is there any literature for the theory of linear programming over affine spaces?

Reply:
In my experience it's uncommon to require that the objective be zero at the origin; especially since often when a function is described as "linear" it really means "affine"

Reply:
Aren't we talking about two different things at the same time?

The first thing, which is a pretty small change, is to allow affine functions in addition to linear functions in the objective.

The second thing, which is quite nontrivial, is to deny that the space of solutions contains some zero solution (which may be infeasible), even if the objective value for zero solution is not required to be zero.

Reply:
Sketch what it could look like with affine stuff:
#10159